### ISSUE TITLE
Authentication block when trying to enable mobile session

### ISSUE THREAD

**Issue by Firepal** [Dniel97/orpheusdl-tidal#32]
**Describe the bug**
Creating a mobile session results in an error during authentication; tidal's response is a "blocked" message.

**To Reproduce**
 1. Do something with the module `python orpheus.py search tidal "any"`
 2. Link to Tidal account, then enter username and password
 3. See error

  **Expected behavior**
The mobile session should be successful

  **Screenshots**
```
PS E:\sources\orpheusdl> python orpheus.py search tidal album "any"

   ____             _                    _____  _
  / __ \           | |                  |  __ \| |
 | |  | |_ __ _ __ | |__   ___ _   _ ___| |  | | |
 | |  | | '__| '_ \| '_ \ / _ \ | | / __| |  | | |
 | |__| | |  | |_) | | | |  __/ |_| \__ \ |__| | |____
  \____/|_|  | .__/|_| |_|\___|\__,_|___/_____/|______|
             | |
             |_|


DEBUG:root:Orpheus: Modules detected: example, tidal
DEBUG:root:Orpheus: example added as a module
DEBUG:root:Orpheus: tidal added as a module
 "enable_mobile" is enabled but no MOBILE session was found. Do you want to create a MOBILE session (used for AC-4/360RA) [Y/n]? y
DEBUG:root:TIDAL: TV session found, loading
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): api.tidal.com:443
DEBUG:urllib3.connectionpool:https://api.tidal.com:443 "GET /v1/sessions HTTP/1.1" 200 271
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): api.tidal.com:443
DEBUG:urllib3.connectionpool:https://api.tidal.com:443 "GET /v1/users/<redacted!>/subscription?countryCode=FR HTTP/1.1" 200 280
DEBUG:root:TIDAL: No MOBILE_DEFAULT session found, creating new one
 TIDAL: Creating a Mobile session
 TIDAL: Enter your TIDAL username and password:
 TIDAL: (password will not be echoed)
 Username: any
 Password:
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): login.tidal.com:443
DEBUG:urllib3.connectionpool:https://login.tidal.com:443 "GET /authorize?response_type=code&redirect_uri=https%3A%2F%2Ftidal.com%2Fandroid%2Flogin%2Fauth&lang=en_US&appMode=android&client_id=WAU9gXp3tHhK4Nns&client_unique_key=<redacted!>&code_challenge=<redacted!>&code_challenge_method=S256&restrict_signup=true HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): dd.tidal.com:443
DEBUG:urllib3.connectionpool:https://dd.tidal.com:443 "POST /js/ HTTP/1.1" 200 230
DEBUG:urllib3.connectionpool:https://login.tidal.com:443 "POST /api/email?response_type=code&redirect_uri=https%3A%2F%2Ftidal.com%2Fandroid%2Flogin%2Fauth&lang=en_US&appMode=android&client_id=WAU9gXp3tHhK4Nns&client_unique_key=<redacted!>&code_challenge=<redacted!>&code_challenge_method=S256&restrict_signup=true HTTP/1.1" 403 720
Traceback (most recent call last):
  File "E:\sources\orpheusdl\orpheus.py", line 224, in <module>
    main()
  File "E:\sources\orpheusdl\orpheus.py", line 113, in main
    module = orpheus.load_module(modulename)
  File "E:\sources\orpheusdl\orpheus\core.py", line 210, in load_module
    loaded_module = class_(module_controller)
  File "E:\sources\orpheusdl\modules\tidal\interface.py", line 114, in __init__
    sessions[session_type].auth(username, password)
  File "E:\sources\orpheusdl\modules\tidal\tidal_api.py", line 392, in auth
    raise TidalAuthError(r.text)
modules.tidal.tidal_api.TidalAuthError: {"url":"https://geo.captcha-delivery.com/captcha/?initialCid=<personal data>"}
PS E:\sources\orpheusdl>

```

  **Desktop (please complete the following information):**
  - OS: Windows 10
  - Python version 3.10.8

**Comment by EmmaJaneBonestell** [comment]
I came across a similar issue today when trying to do the same thing. Using Linux rather than Windows, but otherwise the same. Are you still getting this error, or has the response changed to the below?

```
Traceback (most recent call last):
  File "/ia/b/tidal/orpheusdl/orpheus.py", line 224, in <module>
    main()
  File "/ia/b/tidal/orpheusdl/orpheus.py", line 113, in main
    module = orpheus.load_module(modulename)
  File "/ia/b/tidal/orpheusdl/orpheus/core.py", line 210, in load_module
    loaded_module = class_(module_controller)
  File "/ia/b/tidal/orpheusdl/modules/tidal/interface.py", line 114, in __init__
    sessions[session_type].auth(username, password)
  File "/ia/b/tidal/orpheusdl/modules/tidal/tidal_api.py", line 354, in auth
    raise TidalAuthError("TIDAL BOT protection, try again later!")
modules.tidal.tidal_api.TidalAuthError: TIDAL BOT protection, try again later!
```

**Comment by Dniel97** [comment]
Disable `enable_mobile` in the settings.json and just use the TV session for now. Unless Dolby AC-4 or Sony 360RA is required, you shouldn't notice a difference.

**Comment by Firepal** [comment]
Same. Wanted 360RA... Master is still good, thank you for this software.

**Comment by SI0209** [comment]
I am also experiencing same issue on both RedSea and Orpheus

**Comment by Firepal** [comment]
bump for stalebot

**Comment by Dniel97** [comment]
@Firepal I will look into it next week when I have more time.

**Comment by Dniel97** [comment]
@Firepal Just looked into it and is still working perfectly fine on my end. Please make sure to update the TIDAL module to the latest master commit.

**Comment by Dniel97** [comment]
> I came across a similar issue today when trying to do the same thing. Using Linux rather than Windows, but otherwise the same. Are you still getting this error, or has the response changed to the below?
> 
> ```
> Traceback (most recent call last):
>   File "/ia/b/tidal/orpheusdl/orpheus.py", line 224, in <module>
>     main()
>   File "/ia/b/tidal/orpheusdl/orpheus.py", line 113, in main
>     module = orpheus.load_module(modulename)
>   File "/ia/b/tidal/orpheusdl/orpheus/core.py", line 210, in load_module
>     loaded_module = class_(module_controller)
>   File "/ia/b/tidal/orpheusdl/modules/tidal/interface.py", line 114, in __init__
>     sessions[session_type].auth(username, password)
>   File "/ia/b/tidal/orpheusdl/modules/tidal/tidal_api.py", line 354, in auth
>     raise TidalAuthError("TIDAL BOT protection, try again later!")
> modules.tidal.tidal_api.TidalAuthError: TIDAL BOT protection, try again later!
> ```

You could try connecting with a different IP address or use a VPN. It often helps to wait a day and try it again.

**Comment by EmmaJaneBonestell** [comment]
@Dniel97 I have tried every few days since I posted, and I waited a few days before posting in the first place. I really don't think just waiting and trying again is the solution.

I'll give it a try later I suppose, but I'm not sure why I need a VPN when I have my own private network. I'm not trying to do it over mobile data or from a shared network like from school/work.

I am also using fresh git clones of orpheusdl and orpheusdl-tidal.

Since you cannot reproduce the error, I'm curious, what OS and python version are you using? 
My current setup is:
```
Disttibution ID: Ubuntu
Release: 22.04
Codename: jammy

Python Version: 3.10.6
```

Installed package versions from requirements.txt:
```
defusedxml 0.7.1
protobuf 3.15.8
pycryptodomex 3.17
Pillow 9.3.0
requests 2.28.2
tqdm 4.65.0
mutagen 1.46.1
ffmpeg-python 0.2.0
m3u8 3.5.0
```

I additionally use Pillow-SIMD, which is a drop-in replacement of Pillow, but I seriously doubt that is the cause of the issue.

**Comment by EmmaJaneBonestell** [comment]
Okay. So I used a virtual env instead of system packages, and it worked. For the packages it installed that were not in the requirements.txt, I updated one at a time.

Updating urllib3 fixed it; unfortunately, it needs version 2.0.3, and a lot of maintainers have not updated to that. 

Regardless, adding urllib3>=2.0.3 to orpheusdl's requirements.txt will fix this issue.
